src/sql/workbench/contrib/notebook/browser/models/untitledNotebookInput.ts

```diff
@@ -26,6 +26,7 @@ export class UntitledNotebookInput extends NotebookInput {
 		@IExtensionService extensionService: IExtensionService
 	) {
 		super(title, resource, textInput, true, textModelService, instantiationService, notebookService, extensionService);
+		this.setMode('notebook');
```

Comment:
Ok, tested fix and seems good (the issue I was still seeing is what you mentioned has been around for a while where the save dialog file lists the JSON file types).

Just one small change though - do this instead : 

```
// Set the mode explicitly so that the auto language detection doesn't run and mark the model as being JSON
this.textInput.resolve().then(() => this.setMode('notebook'));
```

I noticed while debugging into this that setting the mode immediately like this meant it wasn't actually setting the mode on the underlying model due to this check here : https://github.com/microsoft/azuredatastudio/blob/2bc6a0cd012de69504061a876070cfa43533fedc/src/vs/workbench/common/editor/textEditorModel.ts#L91

Now this didn't really affect your fix because it was still setting the `_hasModeSetExplicitly` property correctly - but I want to avoid potential other issues of having the mode be 'plaintext' (the default).